src/adapters/AxelarAdapter.sol

```diff
@@ -41,22 +41,27 @@ contract AxelarAdapter is Auth, IAxelarAdapter {
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc IAxelarAdapter
-    function file(bytes32 what, string calldata axelarId, uint16 centrifugeId, string calldata source) external auth {
+    function file(bytes32 what, string memory axelarId, uint16 centrifugeId, string memory source) public auth {
         if (what == "sources") sources[axelarId] = AxelarSource(centrifugeId, keccak256(bytes(source)));
         else revert FileUnrecognizedParam();
         emit File(what, axelarId, centrifugeId, source);
     }
 
     /// @inheritdoc IAxelarAdapter
-    function file(bytes32 what, uint16 centrifugeId, string calldata axelarId, string calldata destination)
-        external
-        auth
-    {
+    function file(bytes32 what, uint16 centrifugeId, string memory axelarId, string memory destination) public auth {
         if (what == "destinations") destinations[centrifugeId] = AxelarDestination(axelarId, destination);
         else revert FileUnrecognizedParam();
         emit File(what, centrifugeId, axelarId, destination);
     }
 
+    /// @inheritdoc IConfigurableAdapter
+    function wire(bytes memory encodedParams) external auth {
```

Comment:
Since this addition introduces a breaking change, it appears to me we need to update the Guardian and all adapters on each network once we deploy v3.1, right? Otherwise, v3 adapters don't work with the v3.1 Guardian wiring protocol.

Reply:
Correct, agree that's not ideal

Reply:
I just realized two things:
1. Swapping adapters wil be quite hard ,as any in-flight messages will need to be recovered.
2. We anyways will need to swap adapters as `entrypoint` is `immutable`, and we will be swapping `entrypoint` aka `multiAdapter` for the per-pool adapter setup

Reply:
Correct on both 😅.

Hopefully, pausing the gateway for some time (30 min or so) before the update help with 1, and only very few messages need to be recovered. Also happy that our recovery method is already in the Gateway in the failed message storage, instead of fully rely on the adapters.

Maybe we should modify the immutable endpoint to a mutable one, for future upgrades after v3.1, WDYT?